libs/ledgerjs/packages/cryptoassets/src/tokens.ts

```diff
@@ -140,12 +172,61 @@ function comparePriority(a: TokenCurrency, b: TokenCurrency) {
   return Number(!!b.disableCountervalue) - Number(!!a.disableCountervalue);
 }
 
+function removeElemFromArray(array: TokenCurrency[], token: TokenCurrency) {
+  if (array && array.length > 0) {
+    const index = array.findIndex(
+      (currentToken) => currentToken && token.id === currentToken.id
+    );
+    if (index === -1) return array;
+    return array.splice(index, 1);
+  }
+}
+
+function cleanTokenArray(
+  array: Record<string, TokenCurrency>,
+  stringIndex: string
+) {
+  tokenListHashes.delete(array[stringIndex]);
+  delete array[stringIndex];
+}
+
+/**
+ * Delete previous token entry to all array
+ * @param token
+ */
+function cleanExistingTokenInArray(token: TokenCurrency) {
+  const { id, contractAddress, parentCurrency, ticker } = token;
+  const lowCaseContract = contractAddress.toLowerCase();
+
+  cleanTokenArray(tokensById, id);
+  cleanTokenArray(
+    tokensByCurrencyAddress,
+    parentCurrency.id + ":" + lowCaseContract
+  );
+  cleanTokenArray(tokensByAddress, lowCaseContract);
+  cleanTokenArray(tokensByTicker, ticker);
+  removeElemFromArray(tokensArray, token);
+  removeElemFromArray(tokensArrayWithDelisted, token);
+  removeElemFromArray(tokensByCryptoCurrency[parentCurrency.id], token);
+  removeElemFromArray(
+    tokensByCryptoCurrencyWithDelisted[parentCurrency.id],
+    token
+  );
+}
+
 export function addTokens(list: TokenCurrency[]): void {
   list.forEach((token) => {
-    if (tokensById[token.id]) return;
+    if (!token) return;
+    const tokenHash = createTokenHash(token);
+    if (tokenListHashes.has(tokenHash)) return;
+
+    const { id, contractAddress, parentCurrency } = token;
+    const lowCaseContract = contractAddress.toLowerCase();
+    cleanExistingTokenInArray(token);
```

Comment:
Can you please add a comment here just to explain that we're doing this for the sole purpose of potential updates of token already in the list (maybe mentionning the word dynamic CAL) 🙏